### PR TITLE
Added as_rect(), update() functions to Circle

### PR DESCRIPTION
**Pull request by itzpr3d4t0r**

### CODE REVIEW

File: geometry.pyi [hunk 1]
```diff
@@ -124,3 +124,8 @@ class Circle:
     def collidepoint(self, point: Coordinate) -> bool: ...
     def __copy__(self) -> "Circle": ...
     copy = __copy__
+    def as_rect(self) -> Rect: ...
+    @overload
+    def pg_circle_update(self, circle: CircleValue) -> None: ...
```
Comment (Emc2356) [inline geometry.pyi:129]: too much copy paste

File: geometry.pyi [hunk 2]
```diff
@@ -124,3 +124,8 @@ class Circle:
...
+    @overload
+    def pg_circle_update(self, x: float, y: float, r: float) -> None: ...
```
Comment (Emc2356) [inline geometry.pyi:131]: too much copy paste

File: src_c/circle.c [hunk 3]
```diff
@@ -262,11 +262,36 @@ pg_circle_collidepoint(pgCircleObject *self, PyObject *const *args,
                  "collidepoint requires a point or PointLike object");
 }
 
+static PyObject *
+pg_circle_as_rect(pgCircleObject *self, PyObject *_null)
+{
+    pgCircleBase scirc = self->circle;
```
Comment (Emc2356) [inline src_c/circle.c:268]: preferably make it `pgCircleBase circle =  &(self->circle)`

File: src_c/circle.c [hunk 4]
```diff
@@ -262,11 +262,36 @@ pg_circle_collidepoint(pgCircleObject *self, PyObject *const *args,
...
+    }
+    Py_RETURN_NONE;
+}
+
 static struct PyMethodDef pg_circle_methods[] = {
     {"collidecircle", (PyCFunction)pg_circle_collidecircle, METH_FASTCALL,
      NULL},
     {"collideline", (PyCFunction)pg_circle_collideline, METH_FASTCALL, NULL},
     {"collidepoint", (PyCFunction)pg_circle_collidepoint, METH_FASTCALL, NULL},
+    {"as_rect", (PyCFunction)pg_circle_as_rect, METH_FASTCALL, NULL},
```
Comment (Emc2356) [inline src_c/circle.c:293]: should be `METH_NOARGS`

File: src_c/circle.c [hunk 5]
```diff
@@ -262,11 +262,36 @@ pg_circle_collidepoint(pgCircleObject *self, PyObject *const *args,
                  "collidepoint requires a point or PointLike object");
 }
 
+static PyObject *
+pg_circle_as_rect(pgCircleObject *self, PyObject *_null)
```
Comment (Emc2356) [inline src_c/circle.c:266]: you can remove `PyObject *_null` as it is `METH_NOARGS` normally

Comment (Starbuck5) [reply src_c/circle.c:266]: It is still of type PyCFunction, even under NoArgs, correct?

we cannot do function pointer casting in pygame as it breaks WASM.

Comment (itzpr3d4t0r) [reply src_c/circle.c:266]: starbuck is right here

**Pull request merged**